Formula/premake.rb

```diff
@@ -17,31 +17,32 @@ class Premake < Formula
   end
 
   # See: https://groups.google.com/g/premake-development/c/i1uA1Wk6zYM/m/kbp9q4Awu70J
-  deprecate! date: "2015-05-28", because: :unsupported
+  deprecate! date: "2015-05-28", because: "premake4 is unsupported and premake5 is still alpha"
 
   def install
     if build.head?
-      system "make", "-f", "Bootstrap.mak", "osx"
-      system "./premake5", "gmake"
-    end
-
-    on_macos do
-      system "make", "-C", "build/gmake.macosx"
-    end
-    on_macos do
-      system "make", "-C", "build/gmake.unix"
-    end
-
-    if build.head?
+      platform = "osx"
+      on_linux do
+        platform = "linux"
+      end
+      system "make", "-f", "Bootstrap.mak", platform
+      system "./bin/release/premake5", "gmake2"
+      system "./bin/release/premake5", "embed"
+      system "make"
       bin.install "bin/release/premake5"
     else
+      platform = "macosx"
+      on_linux do
+        platform = "unix"
+      end
+      system "make", "-C", "build/gmake.#{platform}"
```

Comment:
What's the difference between `gmake.macosx` and `gmake.unix`? It seems using `gmake.unix` seems to work on macOS too. At the very least, it was run when building the ARM bottle (and possibly the Intel Big Sur bottle too).

Reply:
`gmake.macosx` adds flags to link to macOS libraries, e.g. `-framework CoreServices` and `-DLUA_USE_MACOSX`.

I would guess that `gmake.unix` would generally work for most Unix/Linux systems.

Since the bottles were originally built with `gmake.macosx`, I think we should keep that.
The `on_macos` section for `gmake.unix` appears to be a typo from previous linuxbrew backport.

Reply:
Do we know what happens if you use `gmake.unix` after `gmake.macosx`? Because it might be that the ARM bottle is missing the macOS flags.

Reply:
On ARM, it actually fails with current formula, so I think the ARM bottle wasn't built this way:
```
==> make -C build/gmake.macosx
==> make -C build/gmake.unix
Last 15 lines from /Users/cho-m/Library/Logs/Homebrew/premake/02.make:
Undefined symbols for architecture arm64:
  "_CFBundleCopyExecutableURL", referenced from:
      _premake_locate in premake.o
  "_CFBundleGetMainBundle", referenced from:
      _premake_locate in premake.o
  "_CFStringGetCString", referenced from:
      _premake_locate in premake.o
  "_CFURLCopyFileSystemPath", referenced from:
      _premake_locate in premake.o
  "_Gestalt", referenced from:
      _os_getversion in os_getversion.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [../../bin/release/premake4] Error 1
make: *** [Premake4] Error 2
```

Reply:
That's strange, because the typo was merged in November (dc79b87e60f2e2336a3e057327437c09307a749e), but the ARM bottle was built in December (dfd8aab7cf4f24fae06ed0190ef6c7da297ae78a).

Reply:
Not sure why it is behaving differently now. But, I think it was at least a no-op, since ARM bottle is linked to framework:
```console
❯ otool -L bin/premake4
bin/premake4:
	/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 1122.11.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1770.255.0)
```